marklogic-data-hub-spark-connector/spark-test-project/src/main/java/test/StreamTest.java

```diff
@@ -0,0 +1,104 @@
+package test;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SQLContext;
+import org.apache.spark.sql.streaming.StreamingQueryException;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.streaming.Durations;
+import org.apache.spark.streaming.api.java.JavaStreamingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+
+public class StreamTest {
+
+    private static Logger logger = LoggerFactory.getLogger(SparkTest.class);
+
+    private static String host = "localhost";
+    private static String username = "pari";
+    private static String password = "password";
+
+    public static void main(String[] args) throws Exception {
+        setConnectionProperties(args);
+
+        logger.info("Creating JavaSparkContext");
+
+        SparkConf sparkConf = new SparkConf()
+            .setAppName("TestAppName")
+            .setMaster("local[1]")
+            .set("spark.driver.allowMultipleContexts", "true");
+
+        JavaStreamingContext streamingContext = new JavaStreamingContext(sparkConf, Durations.seconds(1));
+        JavaSparkContext javaSparkContext = streamingContext.sparkContext();
+
+        try {
+            Dataset<Row> rows = loadRowsFromTestFile(javaSparkContext);
+            rows.join(rows);
+            writeRowsToDataHub(rows);
+        } finally {
+            logger.info("Closing JavaSparkContext");
+            javaSparkContext.close();
+        }
+    }
+
+    private static void setConnectionProperties(String[] args) {
+        if (args.length != 3) {
```

Comment:
Just a note, this duplication is totally fine right now. I'll submit a PR later to refactor it.